qiskit_experiments/library/hamiltonian/heat_base.py

```diff
@@ -0,0 +1,250 @@
+        """Create new HEAT sub experiment.
+
+        Args:
+            qubits: Index of control and target qubit, respectively.
+            prep_circ: A circuit to prepare qubit before the echo sequence.
+            echo_circ: A circuit to selectively amplify the specific error term.
+            meas_circ: A circuit to project target qubit onto the basis of interest.
+            backend: Optional, the backend to run the experiment on.
+            parameter_name: A name of :math:`d\\theta` parameter from the
+                amplification fit. The fit parameter is represented by this name
+                in the analysis result.
+
+        Keyword Args:
+            See :meth:`experiment_options` for details.
+        """
+        analysis = HeatElementAnalysis()
+        analysis.set_options(result_parameters=[ParameterRepr("d_theta", parameter_name, "rad")])
+
+        super().__init__(qubits=qubits, backend=backend, analysis=analysis)
+        self.set_experiment_options(**kwargs)
+
+        # These are not user configurable options. Be frozen once assigned.
+        self._prep_circuit = prep_circ
+        self._echo_circuit = echo_circ
+        self._meas_circuit = meas_circ
+
+    @classmethod
+    def _default_experiment_options(cls) -> Options:
+        """Default experiment options.
+
+        Experiment Options:
+            repetitions (Sequence[int]): A list of the number of echo repetitions.
+            heat_gate (Gate): A gate instance representing the entangler sequence.
+        """
+        options = super()._default_experiment_options()
+        options.repetitions = list(range(21))
+        options.heat_gate = circuit.Gate("heat", num_qubits=2, params=[])
+
+        return options
+
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        """Default transpile options."""
+        options = super()._default_transpile_options()
+        options.basis_gates = ["sx", "x", "rz", "heat"]
+        options.optimization_level = 1
+
+        return options
+
+    def circuits(self) -> List[QuantumCircuit]:
+        opt = self.experiment_options
+
+        circs = list()
+        for repetition in opt.repetitions:
+            circ = circuit.QuantumCircuit(2, 1)
+            circ.compose(self._prep_circuit, qubits=[0, 1], inplace=True)
+            circ.barrier()
+            for _ in range(repetition):
+                circ.append(self.experiment_options.heat_gate, [0, 1])
+                circ.compose(self._echo_circuit, qubits=[0, 1], inplace=True)
+                circ.barrier()
+            circ.compose(self._meas_circuit, qubits=[0, 1], inplace=True)
+            circ.measure(1, 0)
+
+            # add metadata
+            circ.metadata = {
+                "experiment_type": self.experiment_type,
+                "qubits": self.physical_qubits,
+                "xval": repetition,
+            }
+
+            circs.append(circ)
+
+        return circs
+
+
+class BatchHeatHelper(BatchExperiment, ABC):
```

Comment:
You don't need to act on this comment in this PR, but it might be worth integrating this behavior into the batch_experiment.py, either in `BatchExperiment` or a subclass. There is not really anything significant here that is specific to HEAT (the default options could move to `ZXHeat`). I think it is okay to leave things as they are, but it would be nice to eliminate the `BatchHeatHelper` class just because there are a lot of `Heat` classes added in this PR and it can be tricky to keep track of them when first looking into the HEAT code.

Reply:
This is kind of this experiment specific, since general batch can take experiments that are expected to be transpiled with different transpile options. As you mentioned, probably we can turn this into another baseclass to support option propagation, or, adding decorator to do that.